src/service/viewport-impl.js

```diff
@@ -444,26 +450,100 @@ export class Viewport {
 
   /**
    * Instruct the viewport to enter lightbox mode.
+   * Requesting element is necessary to be able to enter lightbox mode under FIE
+   * cases.
+   * @param {!Element=} opt_requestingElement
    * @return {!Promise}
    */
-  enterLightboxMode() {
+  enterLightboxMode(opt_requestingElement) {
     this.viewer_.sendMessage('requestFullOverlay', {}, /* cancelUnsent */true);
     this.enterOverlayMode();
     this.hideFixedLayer();
+
+    if (opt_requestingElement) {
+      this.maybeEnterFieLightboxMode(
+          dev().assertElement(opt_requestingElement));
+    }
+
     return this.binding_.updateLightboxMode(true);
   }
 
   /**
    * Instruct the viewport to leave lightbox mode.
+   * Requesting element is necessary to be able to enter lightbox mode under FIE
+   * cases.
+   * @param {!Element=} opt_requestingElement
    * @return {!Promise}
    */
-  leaveLightboxMode() {
+  leaveLightboxMode(opt_requestingElement) {
     this.viewer_.sendMessage('cancelFullOverlay', {}, /* cancelUnsent */true);
     this.showFixedLayer();
     this.leaveOverlayMode();
+
+    if (opt_requestingElement) {
+      this.maybeLeaveFieLightboxMode(
+          dev().assertElement(opt_requestingElement));
+    }
+
     return this.binding_.updateLightboxMode(false);
   }
 
+  /**
+   * @return {boolean}
+   * @visibleForTesting
+   */
+  isLightboxExperimentOn() {
+    return isExperimentOn(this.ampdoc.win, A4A_LIGHTBOX_EXPERIMENT);
+  }
+
+  /**
+   * Enters frame lightbox mode if under a Friendly Iframe Embed.
+   * @param {!Element} requestingElement
+   * @visibleForTesting
+   */
+  maybeEnterFieLightboxMode(requestingElement) {
+    this.withFriendlyIframeEmbed_(requestingElement, fie => {
+      dev().assert(this.isLightboxExperimentOn(),
+          `Lightbox mode for A4A is only available when ` +
+          `'${A4A_LIGHTBOX_EXPERIMENT}' experiment is on`);
+
+      fie.enterFullOverlayMode();
+    });
+  }
+
+  /**
+   * Leaves frame lightbox mode if under a Friendly Iframe Embed.
+   * @param {!Element} requestingElement
+   * @visibleForTesting
+   */
+  maybeLeaveFieLightboxMode(requestingElement) {
+    this.withFriendlyIframeEmbed_(requestingElement, fie => {
+      fie.leaveFullOverlayMode();
+    });
+  }
+
+  /**
+   * Execute callback with FriendlyIframeEmbed if available.
+   * @param {!Element} element Element supposedly inside the FIE.
+   * @param {!Function} callback Callback to be executed with FIE as param.
+   * @private
+   */
+  withFriendlyIframeEmbed_(element, callback) {
```

Comment:
Usage is a little less verbose with the callback, but I don't feel strongly about either way. Changed.